shared/controlflow/codeql/controlflow/Cfg.qll

```diff
@@ -1192,6 +1193,78 @@ module Make<LocationSig Location, InputSig<Location> Input> {
     }
   }
 
+  /** Provides the input to `ViewCfgQuery`. */
+  signature module ViewCfgQueryInputSig<FileSig File> {
+    /** The source file selected in the IDE. Should be an `external` predicate. */
+    string selectedSourceFile();
+
+    /** The source line selected in the IDE. Should be an `external` predicate. */
+    string selectedSourceLine();
+
+    /** The source column selected in the IDE. Should be an `external` predicate. */
+    string selectedSourceColumn();
```

Comment:
A relatively recent version of CodeQL should be able to deal with non-string types for external predicates. In any case we should have a sensible type for lines and column numbers in the signature module.

```suggestion
    int selectedSourceLine();

    /** The source column selected in the IDE. Should be an `external` predicate. */
    int selectedSourceColumn();
```

Reply:
Not sure; then all languages would have to make the `.toInt` calls.

Reply:
Oh, I thought we would have to make changes to whatever logic constructs data for `selectedSourceLine` as well.